File: py-codellama34B-AWQ-all/taskid-59_largest_prime_factor-gen14.py
def largest_prime_factor(n: int) -> int:
    """Return the largest prime factor of n. Assume n > 1 and is not a prime.
    >>> largest_prime_factor(13195)
    29
    >>> largest_prime_factor(2048)
    2
    """

    # This is guaranteed to return at least a factor of 2
    i = 2
    while i * i <= n:
        while n % i == 0:
            n //= i
        i += 1
    # If we got here, then n must be prime
    return n if n > 1 else i - 1

File: java-codellama7B-FP-all/taskid-59_largest_prime_factor-gen14.java
import java.util.*;
import java.lang.reflect.*;
import org.javatuples.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    // Return the largest prime factor of n. Assume n > 1 and is not a prime.
    // >>> largestPrimeFactor((13195l))
    // (29l)
    // >>> largestPrimeFactor((2048l))
    // (2l)
    public static long largestPrimeFactor(long n) {

        long n1 = n;
        while (n1 % 2 == 0) {
            n1 /= 2;
        }
        for (long i = 3; i <= Math.sqrt(n1); i += 2) {
            while (n1 % i == 0) {
                n1 /= i;
            }
        }
        if (n1 > 1) {
            return n1;
        } else {
            return n;
        }
    }
    public static void main(String[] args) {
    assert(largestPrimeFactor((15l)) == (5l));
    assert(largestPrimeFactor((27l)) == (3l));
    assert(largestPrimeFactor((63l)) == (7l));
    assert(largestPrimeFactor((330l)) == (11l));
    assert(largestPrimeFactor((13195l)) == (29l));
    }

}
